### DISCUSSION TITLE
Picture Handling: Use Archive to conveniently manage second priority pictures

### DISCUSSION THREAD

**Discussion by MiSebCo** [photoprism/photoprism#4096] [Ideas]
**Describe what problem this solves and why this would be valuable to many users**

In order to keep my library lean but also keeping second priority pictures easily accessible as a user I can use the archive option to conveniently manage second priority pictures. 

Use Case: 
- I have a lot of pictures from a large photo session. I took several pictures from e.g. a group of people or some event. I deleted the bad ones (e.g. with eyes closed, etc.) and now have a selection of good ones. Only best should go into the library but I also do not want to loose the other ones (second priority, at least at the time of decision). I also do not want to be flooded by the second priority pictures in every search, however. So I move the second priority into the archive.

- To have convenient and easy access to the whole session, I put every picture of the session into an album (also the ones from the archive). 

- When the archive is easily accessible users can always go through the whole session to search for pictures they might miss from the event.

- With the album containing the archived pictures it is also possible now to give other people easy access to the whole session which might not even be users of my library, but were part of the event, for example.

- Other users from the library who want to have access to pictures from the event can easily find them in the album, while users searching through existing pictures are not bothered.

- When searching the library users can select to also or only search the archive to be able to use all photoprism search features also with archived pictures. 

**Describe the solution you'd like**

Feature request is based on build 231218-c6cad4614.

Acceptance Criteria:

- [ ] Albums have an option also also show pictures from the archive (similar to "Favorite" and "Private"). This is also visible as icon on the album cover to avoid confusion.
- [ ] Pictures which are in the archive and show up in search results have an archive icon to avoid confusion.
- [ ] There is an option in the search dialogs to additionally go through the archive when searching for pictures. This can easily be turned on an off (like the hidden faces button).
- [ ] There is also an option in the search dialogs to only go through the archive when searching for pictures. This does not need to be accessible on top level.
- [ ] The archive menu has all options (like e.g. adding a picture to an album) - no special menu for the archive. 
- [ ] When an archived picture is added to an album the album feature to show archive pictures becomes active automatically.
- [ ] A new trash is available to distinguish from archive and safeguard from accidentally deleting a file. Files from trash never show up in any search (like the current archive).
- [ ] The archive has no more delete option but a move to trash option. The button move to trash is added also in the other menus.
- [ ] When uploading pictures you can directly upload to the archive (this way you can have an on-boarding stage). The album which is selected gets the archive feature enabled.
- [ ] Whenever archived pictures are selected the option to "unarchive" is available in the menu.

As a note: Similarly to "Archive" there could be an "On-boarding" section for pictures which are currently in the process of being on-boarded to the library but still need pruning before being added to the overall search results. This would separate archive and on-boarding workflows.

**Describe alternatives you've considered**

Stacking of second priority pictures is an option - however this seems to be less intuitive and also does not allow all options as described above.

**Comment by graciousgrey** [comment]
Thank you very much for your feedback!

I do understand the need to be able to flag photos e.g. as "second priority" and to be able to filter for those flags in search and within albums. However our users do have different workflows and using the archive for this might not be the best way to solve this. 

The following feature request might be more suitable:

- https://github.com/photoprism/photoprism/issues/656

It would allow to add custom flags to photos and to filter for them (show/hide). Unlike archived photos, photos with a custom flag could be part of albums and could be found in the normal search. 

Regarding Onboarding, we do already have the review section. At this time only low quality and non-photographic content appears in review. However we might add an option to review all newly added photos.

- https://github.com/photoprism/photoprism/issues/3614

**Comment by MiSebCo** [comment]
Thank you for the feedback, suggestions and crosschecking existing issues and ideas! This is really valuable! Great work! 

https://github.com/photoprism/photoprism/issues/656 could be a very generic way of implementing the idea. This could also enable generating specific workflows for various use cases. Drawback with https://github.com/photoprism/photoprism/issues/656 in my opinion is, that the flagged items will always show up in the search results unless you explicitly suppress them by using the flag to filter out the pictures. 

This could be addressed by adding some properties to the flags in https://github.com/photoprism/photoprism/issues/656, which would also distinguish flags from labels better:
- Flags can already be selected during upload.
- Flags can be used to exclude items in search results and default views (but not from albums) by default.
- Flags can be added to the menu (like Scans, Panorama, Archive) under the search menu.
- Items can be flagged using the context menu in the same way as adding items to an album.
- Flags get a pictogram (like favorite or private).
- A flag can have the following properties (each flag individual)
    - Exclude from search by default
        - Items having this flag do not show up in search results and default views up unless the user explicitly selects it.
    - Show Flag in SideBar
        - The flag shows up as an additional menu Item in die sidebar (like Scans, Panorama, Archive) under the search menu.
    - Flag is set as default when uploading
        - The flag is set by default during upload.
    - Flag shows up as pictogram 
        - User can see the flag directly on the picture.
         
This would also address the problem that archived items cannot be added to albums. 

I will add this comment also to https://github.com/photoprism/photoprism/issues/656

I did not see the #3604. Sorry for that. On the other hand, I would not mix the onboarding workflow with the "to be reviewed" workflow and think the onboarding workflow should be configurable during upload. I personally would not want to push all pictures through the "to be reviewed" stage as this can be a lot of double work. I also added this as a comment to #3604. Flags with properties could make this configurable.

On the other hand, well-documented default workflows for onboarding and archiving in Photoprism could be an added value in my opinion. Currently for me the archive is more a protection from accidently deleting an item. I am searching more for a long-term archive that keeps items I do not want to delete in a separate context. 

It might also be easier for the user to understand the different contexts (onboarding, main library, archive, trash) and it could be ensured that an item is only associated to one context.
